SetTheory/set-relation.R

```diff
@@ -293,175 +293,187 @@ getAsymDiffMC <- function(numSets=2, setSize=5, dType = 1, difficulty = 1) {
   
   #distractor 3 is the original set which is not the complement and is wrong
-  d3 <- sourceSet
+  d3 <- sourceSets[[1]]
   
-  wrongs <- list()
-  wrongs[[1]] <- d1
-  wrongs[[2]] <- d2
-  wrongs[[3]] <- d3
+  distractors[[1]] <- formatListAsSet(d1)
+  distractors[[2]] <- formatListAsSet(d2)
+  distractors[[3]] <- formatListAsSet(d3)
   
-  #format wrong answers as strings
+  
+  
+  #Iterate through the sourceSets. format list as Set and insert at the index.
   counter <- 1
-  for (w in wrongs){
-    current <- formatListAsSet(w)
-    wrongs[counter] <- current
+  for (s in sourceSets){
+    sourceSets[counter] <- formatListAsSet(s)
     counter <- counter + 1
   }
   
-  #format source set as string
-  current <- formatListAsSet(sourceSet)
-  
-  
-  answer <- formatListAsSet(answer)
-  
-  #inserting string formatting "Let A = ...."
-  sourceSet <- insertSetQStrings(sourceSet)
-  
-  #The actual question being asked of the sets. 
-  questionStr <- "Let A be a single set. What is the complement of set A?"
-  
-  #combining the source sets and question string.
-  sourceSet <- c(questionStr, sourceSet)
+  #format the the sourceSets as Question Strings
+  # "A = {...}"
+  # "B = {...}"
+  sourceSets <- insertSetQStrings(sourceSets)
   
-  #format answers and sources into json and return results 
-  toSend <- list(content= sourceSet, correct= answer, distractors= wrongs)
+  # now we concatenate the question contents together
+  questionContents <- c(questionText, sourceSets)
   
+  #add all items to a list for return
+  toSend <- list(content = questionContents, correct = correct, distractors = distractors)
   
+  #return question info
   return(toSend)
-  
 }
 
-getSetEqualityMC <- function(n=2, m = 5, dType = 1) {
-  n <-2   #currently, the api only supports 2 sets. 
   
-  sourceSets <- list()    #creating an empty list. Elements will be sets 
+# getSetEquality() generates and prepares 2 sets of m members 
+# as well as 1 "distractor" and 1 correct answer.
+# The correct answer is a string which states whether the sets are equal
+# or not.
+#
+#
+# @param  numSets        The number of sets to consider
+# @param  setSize        The number of elements in each set. 
+# @response json         A json object containing the
+#                        sets, correct, and incorrect
+#                        answers.
+#
+
+getSetEqualityMC <- function(numSets = 2, setSize = 5, dType = 1) {
+  questionText <- "Let A and B be two sets. Are A and B equal?"
   
-  #for each in a given number of sets, fill the set with ints (1:9)
-  for(e in (1:n)) {
-    sourceSets[[e]] <- sample(1:9, m, replace = F)
-  }
-  allElements <- vector()
+  #generate and fill sets
+  sourceSets <- getSets(n = numSets, m = setSize, x = dType)
   
-  for(e in sourceSets) {
-    print(e)
-    allElements <- c(allElements, e)
+  #sets 50/50 probability of generated sets being equal or not.
+  probability = sample(1:2, 1, replace = FALSE)
```

Comment:
This (line 385) isn't wrong, but it does not  adhere to our coding conventions. Use the left assignment operator "<-" when possible.

Reply:
Note: I'm referring to "probability = sample(..." and not "replace = False"